Lib/modulefinder.py

```diff
@@ -93,7 +94,12 @@ def _find_module(name, path=None):
     else:  # Should never happen.
         return None, None, ("", "", _SEARCH_ERROR)
 
-    file = open(file_path, mode)
+    encoding = None
+    if 'b' not in mode:
+        with open(file_path, 'rb') as file:
+            encoding = tokenize.detect_encoding(file.readline)[0]
+
+    file = open(file_path, mode, encoding=encoding)
```

Comment:
maybe slightly simpler?

```python
if mode == 'r':
    file = tokenize.open(file_path)
else:
    file = open(file_path, mode)
```

Reply:
Sure, I'll update it later today. Thanks!

Reply:
I just added your suggestion and all test passed locally on my computer. Thanks!